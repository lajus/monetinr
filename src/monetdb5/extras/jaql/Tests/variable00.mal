jaql.x("A = [1,2,4,{\"a\": 5}];");
jaql.x("A;");
jaql.x("A -> filter $ <= 2;");
jaql.x("A -> filter each v v.a >= 5;");
jaql.x("n = [ [3,65,8,72], [5,98,2,65] ];");
jaql.x("n -> expand -> transform each n (n * 2);");
jaql.x("n -> transform each n (n);");
jaql.x("n -> transform each n (n[2]);");
jaql.x("n -> transform each n (n[8]);");
jaql.x("n -> transform each n (n[*]);");
jaql.x("m = [ {\"a\":[3,65,{\"b\":8},72]}, {\"a\":[5,{\"b\":98},2,65]} ];");
jaql.x("m -> transform each n (n.a[1]);");
jaql.x("m -> transform each n (n.a[*]);");
jaql.x("m -> transform each n (n.a[*].b);");
jaql.x("m -> transform each n (n.a[2].b);");
jaql.x("o = [ [{\"a\":3},{\"a\":65}], [{\"a\":5},{\"a\":98}] ];");
jaql.x("o -> transform each o (o);");
jaql.x("o -> transform each o (o[1].a);");
jaql.x("o -> transform each o (o[2].a);");
jaql.x("o -> transform each o (o[*].a);");
