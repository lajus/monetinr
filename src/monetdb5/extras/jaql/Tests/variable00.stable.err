stderr of test 'variable00` in directory 'monetdb5/extras/jaql` itself:


# 21:12:04 >  
# 21:12:04 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "gdk_dbfarm=/net/sofia.ins.cwi.nl/export/scratch1/fabian/tmp/mtest-jacqueline-sofia.ins.cwi.nl/five/dbfarm" "--set" "mapi_open=true" "--set" "mapi_port=30179" "--set" "monet_prompt=" "--trace" "--forcemito" "--set" "mal_listing=2" "--dbname=mTests_extras_jaql" "variable00.mal"
# 21:12:04 >  

# builtin opt 	gdk_dbname = demo
# builtin opt 	gdk_dbfarm = /ufs/fabian/scratch/ssd/monetdb/jacqueline/program-x86_64/var/lib/monetdb5/dbfarm
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_alloc_map = no
# builtin opt 	gdk_vmtrim = yes
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	gdk_dbfarm = /net/sofia.ins.cwi.nl/export/scratch1/fabian/tmp/mtest-jacqueline-sofia.ins.cwi.nl/five/dbfarm
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 30179
# cmdline opt 	monet_prompt = 
# cmdline opt 	mal_listing = 2
# cmdline opt 	gdk_dbname = mTests_extras_jaql

# 21:12:04 >  
# 21:12:04 >  "Done."
# 21:12:04 >  

