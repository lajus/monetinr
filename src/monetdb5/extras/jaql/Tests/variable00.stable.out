stdout of test 'variable00` in directory 'monetdb5/extras/jaql` itself:


# 21:12:04 >  
# 21:12:04 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "gdk_dbfarm=/net/sofia.ins.cwi.nl/export/scratch1/fabian/tmp/mtest-jacqueline-sofia.ins.cwi.nl/five/dbfarm" "--set" "mapi_open=true" "--set" "mapi_port=30179" "--set" "monet_prompt=" "--trace" "--forcemito" "--set" "mal_listing=2" "--dbname=mTests_extras_jaql" "variable00.mal"
# 21:12:04 >  

# MonetDB 5 server v11.8.0 "jacqueline-65159548356e+"
# Serving database 'mTests_extras_jaql', using 8 threads
# Compiled for x86_64-pc-linux-gnu/64bit with 64bit OIDs dynamically linked
# Found 15.662 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2013 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://sofia.ins.cwi.nl:30179/
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
# MonetDB/DataCell module not loaded: MALException:jaql.context:JAQL environment not found
function user.main():void;
    jaql.x("A = [1,2,4,{\"a\": 5}];");
    jaql.x("A;");
    jaql.x("A -> filter $ <= 2;");
    jaql.x("A -> filter each v v.a >= 5;");
    jaql.x("n = [ [3,65,8,72], [5,98,2,65] ];");
    jaql.x("n -> expand -> transform each n (n * 2);");
    jaql.x("n -> transform each n (n);");
    jaql.x("n -> transform each n (n[2]);");
    jaql.x("n -> transform each n (n[8]);");
    jaql.x("n -> transform each n (n[*]);");
    jaql.x("m = [ {\"a\":[3,65,{\"b\":8},72]}, {\"a\":[5,{\"b\":98},2,65]} ];");
    jaql.x("m -> transform each n (n.a[1]);");
    jaql.x("m -> transform each n (n.a[*]);");
    jaql.x("m -> transform each n (n.a[*].b);");
    jaql.x("m -> transform each n (n.a[2].b);");
    jaql.x("o = [ [{\"a\":3},{\"a\":65}], [{\"a\":5},{\"a\":98}] ];");
    jaql.x("o -> transform each o (o);");
    jaql.x("o -> transform each o (o[1].a);");
    jaql.x("o -> transform each o (o[2].a);");
    jaql.x("o -> transform each o (o[*].a);");
end main;
[ 1, 2, 4, { "a": 5 } ]
[ 1, 2 ]
[ { "a": 5 } ]
[ 6, 130, 16, 144, 10, 196, 4, 130 ]
[ [ 3, 65, 8, 72 ], [ 5, 98, 2, 65 ] ]
[ 8, 2 ]
[ null, null ]
[ [ 3, 65, 8, 72 ], [ 5, 98, 2, 65 ] ]
[ 65, { "b": 98 } ]
[ [ 3, 65, { "b": 8 }, 72 ], [ 5, { "b": 98 }, 2, 65 ] ]
[ [ 8 ], [ 98 ] ]
[ 8, null ]
[ [ { "a": 3 }, { "a": 65 } ], [ { "a": 5 }, { "a": 98 } ] ]
[ 65, 98 ]
[ null, null ]
[ [ 3, 65 ], [ 5, 98 ] ]

# 21:12:04 >  
# 21:12:04 >  "Done."
# 21:12:04 >  

